sql/analyzer/unnest_insubqueries.go

```diff
@@ -206,10 +225,35 @@ func disambiguateTables(used map[string]int, n sql.Node) (sql.Node, error) {
 			})
 		}
 	})
-	for k, v := range duplicates {
-		used[k] = v
+	if err != nil {
+		return nil, nil, err
+	}
+	if len(rename) > 0 {
+		n, _, err = renameExpressionTables(n, rename)
 	}
-	return n, err
+	return n, used, err
+}
+
+// renameExpressionTables renames table references recursively. We use
+// table ids to avoid improperly renaming tables in lower scopes with the
+// same name.
+func renameExpressionTables(n sql.Node, rename map[sql.TableId]string) (sql.Node, transform.TreeIdentity, error) {
```

Comment:
This seems fine but I'm a little confused why this should be necessary. Isn't the point of table IDs so that we can uniquely identify every table without doing string matching?

Reply:
`assignExecIndexes` still does string matching right now, but in general it would be nice to expand unique id use